src/procedure/nodes/iteratedata1d.cpp

```diff
@@ -0,0 +1,112 @@
+                                                       ProcedureNode::NodeContext forEachContext)
+    : ProcedureNode(ProcedureNode::NodeType::IterateData1D, {ProcedureNode::AnalysisContext}), sourceIntegerData_(intTarget),
+      forEachBranch_(forEachContext, *this, "ForEach")
+{
+    setUpKeywords();
+}
+
+// Set up keywords for node
+void IterateData1DProcedureNode::setUpKeywords()
+{
+    keywords_.setOrganisation("Options", "Source");
+    keywords_.add<NodeKeyword<Collect1DProcedureNode>>("SourceData", "Collect1D node containing the histogram data to process",
+                                                       sourceData_, this, ProcedureNode::NodeType::Collect1D, false);
+    keywords_.add<NodeKeyword<IntegerCollect1DProcedureNode>>(
+        "SourceIntegerData", "IntegerCollect1D node containing the histogram data to process", sourceIntegerData_, this,
+        ProcedureNode::NodeType::IntegerCollect1D, false);
+    keywords_.addHidden<NodeBranchKeyword>("ForEach", "Branch to run on each site selected", forEachBranch_);
+}
+
+/*
+ * Branch
+ */
+// Return the branch from this node (if it has one)
+OptionalReferenceWrapper<ProcedureNodeSequence> IterateData1DProcedureNode::branch() { return forEachBranch_; }
+
+/*
+ * Execute
+ */
+
+// Prepare any necessary data, ready for execution
+bool IterateData1DProcedureNode::prepare(const ProcedureContext &procedureContext)
+{
+    if (sourceData_ && sourceIntegerData_)
+    {
+        return Messenger::error("Specify either SourceData or SourceIntegerData, not both.\n");
+    }
+    else if (sourceData_ || sourceIntegerData_)
+    {
+        // If one exists, prepare the ForEach branch nodes
+        if (!forEachBranch_.prepare(procedureContext))
+            return false;
+        else
+            return true;
+    }
+    else
+        return Messenger::error("No source data node set in '{}'.\n", name());
+}
+
+bool IterateData1DProcedureNode::execute(const ProcedureContext &procedureContext)
+{
+    // If a ForEach branch has been defined, process it for each of our sites in turn. Otherwise, we're done.
+    if (!forEachBranch_.empty())
+    {
+        // Retrieve / realise the normalised data from the supplied list
+        auto &data = procedureContext.dataList().realise<Data1D>(
+            fmt::format("Process1D//{}", name()), procedureContext.processingDataPrefix(), GenericItem::InRestartFileFlag);
+        data.setTag(name());
+
+        // Copy the averaged data from the associated Process1D node
+        if (instantaneous_)
+            data = sourceData_ ? sourceData_->data() : sourceIntegerData_->data();
+        else
+            data = sourceData_ ? sourceData_->accumulatedData() : sourceIntegerData_->accumulatedData();
+
+        for (auto index : data.values())
+        {
+            xParameter_->setValue(data.xAxis(index));
+            valueParameter_->setValue(data.value(index));
```

Comment:
This loop is actually wrong - it's iterating over the actual values of the data, rather than the indices of the vectors (that would have needed `data.nPoints()`. Anyway, we can use our nice `zip` function here to iterate simultaneously over the two parts:
```suggestion
        for (const auto &&[x, value] : zip(data.xAxis(), data.values()))
        {
            xParameter_->setValue(x);
            valueParameter_->setValue(value);
```